docs/source/markdown/podman-manifest-push.1.md.in

```diff
@@ -42,6 +42,10 @@ the list or index itself. (Default true)
 
 @@option digestfile
 
+#### **--force-compression**
+
+Use the specified compression algorithm even if the destination contains a differently-compressed variant already.
```

Comment:
I think we need to make this less abstract.  Why do I need that instead of `--compression`?  What's the use case?

Reply:
Same applies to other locations where the functionality is described.

Reply:
How about if i add a text like
```
Use the --force-compression option to force re-compressing layers imported from a previous image, if the requested compression algorithm is different from the previous compression algorithm
```

Reply:
But shouldn't --compression just imply the force part? If I say `--compression zstd` it should do that and not reuse gzip blobs IMO. I cannot think of anyone actually wanting the current behaviour, to me this is a bug and does not need yet another compression option!

Reply:
@Luap99 This is one of the behavior which is already there in `docker` and afaik `c/image` follows it. `--add-compression` does it internally but I think changing behavior of  `--compression` is breaking. I think following behavior can be changed but I am not aware of all the use-case which could break because of changing behaviour of `--compression` , @mtrmac can provide better explanation about this question.

Reply:
This is a performance trade-off: compressing the layer is very frequently the slowest part of the upload, so reusing the layer which exists, even if it does not have exactly the right format, or even if it is uncompressed, might be the right thing to do, especially for locally-built and locally-run images which are not pulled that many times.

And I think it’s quite possible that enforcing re-compression could end up as a workload-breaking regression for some users who rely on pushes of large data to be fast in this situation. I have no idea how likely it is; we know some users want builds to be very fast, but presumably in the existing gzip-mostly ecosystem algorithm mismatches should be _fairly_ rare.

This applies especially in the default case, when the user does not specify `--compression` on the command line. And, sadly, because EngineConfig.CompressionFormat has a hard-coded default in c/common, c/image can’t currently tell between a caller that uses `podman push` and a user that uses `podman push --compression-format gzip`. So, as far as c/image is concerned, this needs to be a new option, to support the exiting performance characteristics for existing callers like current versions of Podman.

I can see an argument that `podman push --compression-format` should imply `ForceCompressionFormat`. I don’t feel strongly about that, if we are talking about the explicit CLI option, not the `containers.conf` default, and not the built-in default.

I think the argument that compression should _always_ use `ForceCompressionFormat` is reasonable in the abstract, but I don’t think it’s the right choice given historical behavior of both Docker and Podman. But if Podman maintainers decide that’s the desired behavior, I can live with that outcome.

Reply:
Well I am not the expert here I just cannot think of that behaviour being wanted by someone if you think that has value for backwards compatibility then I will not argue against that.

What I complain about though is that the current documentation for --compression-format does not mention this at all. In general I think most of these option should cross reference each other so that users can get the full picture.

Reply:
@Luap99 That is a valid request, I'll add a cross link in `--compression-format` so users can become aware of this. But In this thread I am trying to figure out a text for `--force-compression` as @vrothberg requested.

@Luap99 @vrothberg WDYT about: https://github.com/containers/podman/pull/19640?notification_referrer_id=NT_kwDOAI8y-7I3Mzk5ODM5NjA0OjkzODQ2OTk#discussion_r1295704009

Reply:
> @Luap99 @vrothberg WDYT about: https://github.com/containers/podman/pull/19640

I don't understand the question.  Can you clarify where you're pointing at?

Reply:
@vrothberg I am trying to find a good text for man-page in order to resolve your request here: https://github.com/containers/podman/pull/19640?notification_referrer_id=NT_kwDOAI8y-7I3Mzk5ODM5NjA0OjkzODQ2OTk#discussion_r1295590244

So before I amend the doc, my intention was to finalize the text here:

```
Use the --force-compression option to force re-compressing layers imported from a previous image which was pushed using a default compression format or via --compression, if the requested compression algorithm is different from the previous compression algorithm
```

Reply:
@flouthoc can you do a PS in today's meeting on the issue of `--compression-format` maybe including `--force-compression`?

@rhatdan @giuseppe PTAL
I think we can change that later on as well and don't need to block the PR on it.

> So before I amend the doc, my intention was to finalize the text here:

@flouthoc, I think we need to use very simple words.  In the end it breaks down to the fact that `--compression` may not compress if a matching layer exists on the registry. `--force...` will force compression.`

Reply:
we can argue about backward compatibility, but ignoring the value passed into `--compression-format` seems like a bug to me and not intuitive.  If a user expects the layer to be compressed and ignores the compression format, it is already possible to use `--compress`.  I don't see the value in having three different knobs just to customize the compression format to use.

Reply:
@giuseppe Yes two issues I see are backward compatibility and docker compat, other than these two there are no strong reasons which prevents us to automatically include `--force-compression` on every `--compression-format`.